include/quicly/maxsender.h

```diff
@@ -105,19 +106,28 @@ inline void quicly_maxsender_record(quicly_maxsender_t *m, int64_t value, quicly
     m->max_committed = value;
     ++m->num_inflight;
     m->force_send = 0;
+    sent->inflight = 1;
     sent->value = value;
 }
 
 inline void quicly_maxsender_acked(quicly_maxsender_t *m, quicly_maxsender_sent_t *sent)
 {
     if (m->max_acked < sent->value)
         m->max_acked = sent->value;
-    --m->num_inflight;
+    /* num_inflight should not be adjusted in case of a late ACK */
+    if (sent->inflight) {
+        assert(m->num_inflight != 0);
+        --m->num_inflight;
+        sent->inflight = 0;
+    }
 }
 
 inline void quicly_maxsender_lost(quicly_maxsender_t *m, quicly_maxsender_sent_t *sent)
 {
+    /* the function must be called at most once (when LOST event occurs, but not EXPIRED), hence assert and always decrement */
+    assert(m->num_inflight != 0);
```

Comment:
`assert(m->num_inflight > 0)` maybe?

Reply:
I think I'd prefer keeping the code as-is, due to the following reasons:
* They are the same, due to `num_inflight` being an unsigned type.
* We generally use `numXXX != 0` for checking if something is not empty.
* While I can see that some would think `numXXX > 0` to be more natural, the counterargument that might be compelling to using `!= 0` is that the natural way of testing whether something is empty is doing `numXXX == 0` rather than `numXXX < 0`, and that `numXXX != 0` is the negation of `numXXX == 0`.

I think we should refrain from entering a bikeshed discussion, and the easiest way is to keep our convention as-is (though the ugly thing here is that I am enforcing _my_ preference under the name of "existing convention").